core/controllers/reader.py

```diff
@@ -310,68 +340,107 @@ def post(self, exploration_id):
         Args:
             exploration_id: str. The ID of the exploration.
         """
-        exp_issue_dict = self.payload.get('exp_issue_dict')
-        try:
-            unused_exp_issue = stats_domain.ExplorationIssue.from_backend_dict(
-                exp_issue_dict)
-        except utils.ValidationError as e:
-            raise self.InvalidInputException(e)
-
         playthrough_data = self.payload.get('playthrough_data')
         try:
             unused_playthrough = stats_domain.Playthrough.from_backend_dict(
                 playthrough_data)
         except utils.ValidationError as e:
             raise self.InvalidInputException(e)
 
+        try:
+            issue_schema_version = self.payload['issue_schema_version']
+        except KeyError as e:
+            raise self.InvalidInputException(e)
+
         exp_version = playthrough_data['exp_version']
 
         exp_issues_model = stats_models.ExplorationIssuesModel.get_model(
             exploration_id, exp_version)
         exp_issues = stats_services.get_exp_issues_from_model(exp_issues_model)
 
-        customization_args = exp_issue_dict['issue_customization_args']
-
-        issue_found = False
-        for index, issue in enumerate(exp_issues.unresolved_issues):
-            if issue.issue_type == exp_issue_dict['issue_type']:
-                issue_customization_args = issue.issue_customization_args
-                # In case issue_keyname is 'state_names', the ordering of the
-                # list is important i.e. [a,b,c] is different from [b,c,a].
-                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
-                    issue.issue_type]
-                if (issue_customization_args[issue_keyname] ==
-                        customization_args[issue_keyname]):
-                    issue_found = True
-                    if (len(issue.playthrough_ids) <
-                            feconf.MAX_PLAYTHROUGHS_FOR_ISSUE):
-                        playthrough_id = stats_models.PlaythroughModel.create(
-                            playthrough_data['exp_id'],
-                            playthrough_data['exp_version'],
-                            playthrough_data['issue_type'],
-                            playthrough_data['issue_customization_args'],
-                            playthrough_data['actions'])
+        customization_args = playthrough_data['issue_customization_args']
+
+        # If playthrough already exists, update it in the datastore.
+        if playthrough_data['id']:
+            playthrough_stored = True
+            playthrough_id = playthrough_data['id']
+            playthrough = stats_domain.Playthrough.from_dict(playthrough_data)
+            orig_playthrough = stats_services.get_playthrough_by_id(
+                playthrough_id)
+            if orig_playthrough.issue_type != playthrough.issue_type:
+                # Check whether the playthrough can be added to its new issue,
```

Comment:
Please move the code in this if-branch into a well-named helper method.

Reply:
Done.